src/StreamrClient.js

```diff
@@ -39,54 +45,50 @@ export default class StreamrClient extends EventEmitter {
         this.connection = connection || new Connection(this.options)
 
         // Broadcast messages to all subs listening on stream
-        this.connection.on('b', (msg) => {
+        this.connection.on('BroadcastMessage', (msg) => {
             // Notify the Subscriptions for this stream. If this is not the message each individual Subscription
             // is expecting, they will either ignore it or request resend via gap event.
-            const subs = this.subsByStream[msg.streamId]
+            const subs = this.subsByStream[msg.payload.streamId]
             if (subs) {
                 subs.forEach((sub) => {
-                    sub.handleMessage(msg, false)
+                    sub.handleMessage(msg.payload, false)
                 })
             } else {
-                debug('WARN: message received for stream with no subscriptions: %s', msg.streamId)
+                debug('WARN: message received for stream with no subscriptions: %s', msg.payload.streamId)
             }
         })
 
         // Unicast messages to a specific subscription only
-        this.connection.on('u', (msg, sub) => {
-            if (sub !== undefined && this.subById[sub] !== undefined) {
-                this.subById[sub].handleMessage(msg, true)
+        this.connection.on('UnicastMessage', (msg) => {
+            if (msg.subId !== undefined && this.subById[msg.subId] !== undefined) {
+                this.subById[msg.subId].handleMessage(msg.payload, true)
             } else {
-                debug('WARN: subscription not found for stream: %s, sub: %s', msg.streamId, sub)
+                debug('WARN: subscription not found for stream: %s, sub: %s', msg.payload.streamId, msg.subId)
             }
         })
 
-        this.connection.on('subscribed', (response) => {
-            if (response.error) {
-                this.handleError(`Error subscribing to ${response.stream}: ${response.error}`)
-            } else {
-                const subs = this.subsByStream[response.stream]
-
-                // The typeof array === 'object'
-                if (subs && typeof subs === 'object') {
-                    delete subs.subscribing
-                    // Report subscribed to all non-resending Subscriptions for this stream
-                    subs.filter((sub) => !sub.resending)
-                        .forEach((sub) => {
-                            sub.setState(Subscription.State.subscribed)
-                        })
-                }
+        this.connection.on('SubscribeResponse', (response) => {
+            const subs = this.subsByStream[response.payload.streamId]
 
-                debug('Client subscribed: %o', response)
+            // The typeof array === 'object'
+            if (subs && typeof subs === 'object') {
```

Comment:
If you want to check whether `subs` is an array or not please use `Array.isArray` instead:)
```suggestion
            if (Array.isArray(subs)) {
```
No need for additional nullcheck that way (`Array.isArray(undefined) === false`)